### ISSUE TITLE
CA Modeling Issues

### ISSUE THREAD

**Issue by helencsp** [csp-inc/lumonitor#24]
Working through some sampling issues/questions while running models on CA including:

- [x] - **Trying to pull image chips on model run time without getting all chips from a single image at once**. Took a deeper look last week and randomly iterating images may not be performant enough when compared to pulling a series a single image at once. But it may also be fine. Alternatives I'm exploring include using a BufferedShuffleDataset (sounds like a class name I would come up with) to introduce at least some variety, and balancing the batch size with the chips / tile so it's rare to have a gradient step based solely on a single tile. However, tests indicate this may not be a big issue with model results at the moment.
- [ ] - **Randomly rotating / messing with chips**. Haven't implemented this yet but I think it will be easy through.
- [x] - **Choosing a different label**. So far I've been using the 'Hall' band from the 2017 HM data but I think it has issues, including some features not being visible (e.g. pipelines), edge effects not being visible (though I think the window of most kernels would capture them, that may be dreaming), funny things like some waters being modified and others not, etc. I had considered breaking the inputs down into components but right now I'm just thinking about training w/ NLCD instead (for now at least!). 
- [ ] - Currently **testing regression model with impervious band** but may move to lc bands.
- [ ] - **Changing the model architecture**. Currently using the "stock" UNET from Ronneberger et al 2015 but I know there's likely to be improved models out there. Looked into torchvision models but I'm not sure I can use hyperspectral data with those. Moving from a regression model (since HM is continuous) to a classification model may open up more doors, even though it seems like the final activation function is the only difference.
- [x] - Going to try to **come to terms with what, if anything, is possible before the end of the month**. I'm hopeful in one sense as if I can get

 a model to fit reasonably well I could get results pretty quickly, but that could take anywhere from 1 day to 1 decade.

**Comment by blindjesse** [comment]
Re: first box, I discovered this didn't really seem to be much of an issue, then discovered if I send `shuffle=True` to the dataloader it indeed shuffled files (but not random points within each epoch for some reason). Needs regular "Dataset" though.

Most other issues are moot at this point
